public/app/plugins/panel/graph/graph.ts

```diff
@@ -343,7 +343,8 @@ function graphDirective($rootScope, timeSrv, popoverSrv, contextSrv) {
         eventManager.addFlotEvents(annotations, options);
         configureAxisOptions(data, options);
 
-        sortedSeries = _.sortBy(data, function(series) { return series.zindex; });
+        sortedSeries = _.sortBy(data, (series) => { return series.zindex; });
+        sortSeries(sortedSeries, data, ctrl.panel);
```

Comment:
Looks like it sort twice here (when legend sort is enabled), maybe move zindex sort into sortSeries?

Reply:
Should series override (zindex) or sort as legend rank highest?

I think we should execute zindex sorting at the end.

Reply:
Yes. That might be best. Do it inside sortSeries after any legend sort.